crates/goose/src/providers/formats/openai.rs

```diff
@@ -255,8 +255,17 @@ pub fn create_request(
     tools: &[Tool],
     image_format: &ImageFormat,
 ) -> anyhow::Result<Value, Error> {
+    let mut messages_array = vec![];
+
+    // Determine if the model is "o1", "o1-mini", or something else
+    // Note that o1 is a superset of o1-mini
+    let is_o1 = model_config.model_name.starts_with("o1");
+    let is_o1_mini = model_config.model_name.starts_with("o1-mini");
```

Comment:
i think we can completely skip o1-mini since it doesn't support tools or developer msgs. instead lets put this in documentation here: 
https://github.com/block/goose/blob/6051021a8de7a42859f28f0f8777e93471e790bd/documentation/docs/getting-started/providers.md?plain=1#L23-L27